### ISSUE TITLE
gsettings cannot save settings

### ISSUE THREAD

**Issue by cfusting** [ContinuumIO/anaconda-issues#1479]
Using 14.04 LTS the Conda installed gsettings reports the following error when setting a value:

> GLib-GIO-Message: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.

Workaround is to use the system installed gsettings.

**Comment by moorepants** [comment]
This is still quite an annoying issue. Here is a screenshot of how the cursor hangs on this warning:
![selection_175](https://user-images.githubusercontent.com/276007/27256435-7389b23c-5369-11e7-81f5-f75ee2e08a45.jpg)

and the text:

```
moorepants@garuda:~$ conda info
Current conda install:

               platform : linux-64
          conda version : 4.3.22
       conda is private : False
      conda-env version : 4.3.22
    conda-build version : 2.1.15
         python version : 3.5.3.final.0
       requests version : 2.14.2
       root environment : /home/moorepants/miniconda3  (writable)
    default environment : /home/moorepants/miniconda3
       envs directories : /home/moorepants/miniconda3/envs
                          /home/moorepants/.conda/envs
          package cache : /home/moorepants/miniconda3/pkgs
                          /home/moorepants/.conda/pkgs
           channel URLs : http://repo.continuum.io/pkgs/free/linux-64
                          http://repo.continuum.io/pkgs/free/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/pydy/linux-64
                          https://conda.anaconda.org/pydy/noarch
                          https://conda.anaconda.org/moorepants/linux-64
                          https://conda.anaconda.org/moorepants/noarch
            config file : /home/moorepants/.condarc
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.22 requests/2.14.2 CPython/3.5.3 Linux/4.10.0-22-generic debian/stretch/sid glibc/2.24    
                UID:GID : 1000:1000
moorepants@garuda:~$ ipython
Python 3.5.3 |Continuum Analytics, Inc.| (default, Mar  6 2017, 11:58:13) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.1.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import matplotlib.pyplot as plt

In [2]: GLib-GIO-Message: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
```

Note similar complaints are found in other forums: https://software.intel.com/en-us/forums/intel-distribution-for-python/topic/691231

I've yet to find a fix. It would be nice if this didn't happen. I'm on Ubuntu 17.10.

**Comment by pjgoodall** [comment]
The affects me too

**Comment by everestial** [comment]
Any updates or fix on this issue yet ? Can't seem to figure it out. I am on ubuntu 16.

**Comment by moorepants** [comment]
Here is a related issue with a fix: https://github.com/conda-forge/glib-feedstock/issues/19

**Comment by mingwandroid** [comment]
That looks like a workaround, and one that works only for some applications.

**Comment by jakirkham** [comment]
Indeed. If you have a better idea to solve it @mingwandroid, am all ears. That was a quick workaround that we cobbled together last year to help @moorepants and others to get passed this issue. Unfortunately it doesn't work in all cases and it's a bit dicey hard coding the path given it may vary between distros. Have lacked any time to dig into it further.

**Comment by mingwandroid** [comment]
I won't have time to look into this for a while, sorry.

Can you not use always_include_files for the directory name and maybe put a dummy file in there?

**Comment by improvethings** [comment]
$ gsettings set org.mate.power-manager action-sleep-type-battery suspend
GLib-GIO-Message: 13:39:09.565: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
$ gsettings get org.mate.power-manager action-sleep-type-battery
GLib-GIO-Message: 13:39:11.760: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
'hibernate'

$ which gsettings
/home/me/anaconda3/bin/gsettings
$ /usr/bin/gsettings set org.mate.power-manager action-sleep-type-battery suspend
$ /usr/bin/gsettings get org.mate.power-manager action-sleep-type-battery 
'suspend'
$ gsettings get org.mate.power-manager action-sleep-type-battery 
GLib-GIO-Message: 13:42:03.472: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
'hibernate'

$ uname -srm
Linux 4.9.0-6-amd64 x86_64
$ cat /etc/issue
Debian GNU/Linux 9 \n \l
$ conda info | grep -v home

     active environment : None
 populated config files : 
          conda version : 4.5.4
    conda-build version : 3.10.5
         python version : 3.6.5.final.0
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
               platform : linux-64
             user-agent : conda/4.5.4 requests/2.18.4 CPython/3.6.5 Linux/4.9.0-6-amd64 debian/9 glibc/2.24
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
